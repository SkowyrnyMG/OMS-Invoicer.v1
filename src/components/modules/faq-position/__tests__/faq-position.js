import React from 'react';

import {
  snapShotTest,
  renderWithReduxRouter,
  screen,
  userEvent,
} from 'utils/tests/test-helper';
import FAQPosition from '../faq-position';

describe('FAQPosition', () => {
  test('should display in the document with correct styles', () => {
    snapShotTest(
      <FAQPosition title='TestFAQ title'>TestFAQ description.</FAQPosition>,
    );
  });

  test('should open and close position description on heading click', () => {
    renderWithReduxRouter(
      <FAQPosition title='TestFAQ title'>TestFAQ description.</FAQPosition>,
    );
    let description = screen.queryByText('TestFAQ description.');
    const heading = screen.getByRole('heading', { name: /TestFAQ title/i });

    expect(description).toBeNull();

    userEvent.click(heading);

    description = screen.queryByText('TestFAQ description.');

    expect(description).toBeInTheDocument();
  });
});
