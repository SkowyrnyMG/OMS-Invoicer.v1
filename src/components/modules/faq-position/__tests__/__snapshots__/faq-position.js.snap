// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQPosition should display in the document with correct styles 1`] = `
.c0 {
  padding: 1rem 0.5rem;
  width: 100%;
  min-height: 5rem;
  border-top: 1px solid #446DF644;
}

.c0 img {
  display: block;
  margin: 0 auto;
  width: 50%;
  min-width: 20rem;
}

.c1 {
  width: 100%;
  font-size: 20px;
  color: #8F8F8F;
  cursor: pointer;
}

.c1 * {
  margin: 1rem;
}

<div
  class="c0"
>
  <h5
    class="c1"
  >
    <span>
      +
    </span>
    <span>
      TestFAQ title
    </span>
  </h5>
</div>
`;
